rpcs3/Emu/System.cpp

```diff
@@ -744,7 +752,15 @@ game_boot_result Emulator::BootGame(const std::string& path, const std::string&
 				}
 			}
 		}
+
+		// Restore args manually
+		restore_on_failure(game_boot_result::nothing_to_boot);
 	}
+	else
+	{
+		return restore_on_failure(result);
+	}
+
 	return result;
```

Comment:
why not return result in the if path and return restore_on_failure per default.

Reply:
Fixed.